game/armedforces/armedforces.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+
+import random
+from typing import TYPE_CHECKING, Iterator, Optional
+from game import db
+from game.data.groups import GroupRole, GroupTask
+from game.armedforces.forcegroup import ForceGroup
+from game.profiling import logged_duration
+
+if TYPE_CHECKING:
+    from game.factions.faction import Faction
+
+
+# TODO More comments and rename
+class ArmedForces:
+    """TODO Description"""
+
+    # All available force groups for a specific Role
+    forces: dict[GroupRole, list[ForceGroup]]
+
+    def __init__(self, faction: Faction):
+        with logged_duration(f"Loading armed forces for {faction.name}"):
+            self._load_forces(faction)
+
+    def add_or_update_force_group(self, new_group: ForceGroup) -> None:
+        """TODO Description"""
+        if new_group.role in self.forces:
+            # Check if a force group with the same units exists
```

Comment:
I see the TODOs here about comments and I'm fine with this merging as is, but for some help targeting those comments when you get to the TODOs: my questions here are why duplicates exist, and why overriding is the correct choice as opposed to just appending. I'm guessing it'll be become clear when I have a better understanding of what a `ForceGroup` is, but I haven't gotten there yet :)